### PR TITLE
Hadith of the day

### PR DESCRIPTION
**Pull request by alhabibhasan**
### What does this do?
Introduces a new view, namely one that shows a different hadith (from a list of provided hadith) every day.

The hadith are retrieved from the Google sheets page, https://docs.google.com/spreadsheets/d/1hfVgZonF6On7sS8sDrTZXc8Eak6oH_w7VKLXMSNT1JI/edit?usp=sharing

Feel free to request changes.

### CODE REVIEW

File: src/_components/hadith-data/hadith-data.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import moment from 'moment';
+import config from '../../config.json';
+import csvtojson from 'csvtojson';
+import request from 'request';
+
+export default class HadithData {
+  constructor() {
+    this.getHadithFromGoogleSheets().then(() => {
```
Comment (DilwoarH) [inline src/_components/hadith-data/hadith-data.js]: This causes the app to request the data from GoogleSheets to be called every time the view is show. This should be only done based on the refresh period set.

File: src/_components/hadith-data/hadith-data.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      window.localStorage.setItem('hadithData_currentIndex', currentIndex);
+    }
+  }
+
+  storeHadithData(hadithData = []) {
+    window.localStorage.setItem('hadithData', JSON.stringify(hadithData));
+    window.localStorage.setItem('hadithData_lastUpdated', moment().unix());
+  }
+
+  updateData() {
```
Comment (DilwoarH) [inline src/_components/hadith-data/hadith-data.js]: This function seems like it does what is expected but doesn't seem to be getting called?

**Comment by DilwoarH** [comment]
Also change log needs to be updated, I think this constitutes a new version - 4.0.0 🎊.

https://github.com/DilwoarH/Mosque-Screen/blob/master/CHANGELOG.md
https://github.com/DilwoarH/Mosque-Screen/blob/master/package.json#L3
https://github.com/DilwoarH/Mosque-Screen/blob/master/src/config.json#L2

File: src/_components/hadith-data/hadith-data.js (hunk 1 above)
Comment (alhabibhasan) [reply src/_components/hadith-data/hadith-data.js]: Ah, yh it does, was meant to call the update function in the constructor, I'll change this.

File: src/_components/hadith-data/hadith-data.js (hunk 2 above)
Comment (alhabibhasan) [reply src/_components/hadith-data/hadith-data.js]: Yh, I should be calling it in the constructor.

**Pull request merged**